tedana/tests/test_integrity _robustica.py

```diff
@@ -0,0 +1,667 @@
+
+
+def reclassify_raw_registry() -> str:
+    return os.path.join(reclassify_raw(), "desc-tedana_registry.json")
+
+
+def guarantee_reclassify_data() -> None:
+    """Ensures that the reclassify data exists at the expected path and return path."""
+
+    test_data_path, osf_id = data_for_testing_info("three-echo-reclassify")
+
+    # Should now be checking and not downloading for each test so don't see if statement here
+    # if not os.path.exists(reclassify_raw_registry()):
+    download_test_data(osf_id, test_data_path)
+    # else:
+    # Path exists, be sure that everything in registry exists
+    ioh = InputHarvester(reclassify_raw_registry())
+    all_present = True
+    for _, v in ioh.registry.items():
+        if not isinstance(v, list):
+            if not os.path.exists(os.path.join(reclassify_raw(), v)):
+                all_present = False
+                break
+    if not all_present:
+        # Something was removed, need to re-download
+        shutil.rmtree(reclassify_raw())
+        guarantee_reclassify_data()
+    return test_data_path
+
+
+def test_integration_five_echo(skip_integration):
+    """Integration test of the full tedana workflow using five-echo test data."""
+
+    if skip_integration:
+        pytest.skip("Skipping five-echo integration test")
+
+    test_data_path, osf_id = data_for_testing_info("five-echo")
+    out_dir = os.path.abspath(os.path.join(test_data_path, "../../outputs/five-echo"))
+    # out_dir_manual = f"{out_dir}-manual"
+
+    if os.path.exists(out_dir):
+        shutil.rmtree(out_dir)
+
+    # if os.path.exists(out_dir_manual):
+    #     shutil.rmtree(out_dir_manual)
+
+    # download data and run the test
+    download_test_data(osf_id, test_data_path)
+    prepend = f"{test_data_path}/p06.SBJ01_S09_Task11_e"
+    suffix = ".sm.nii.gz"
+    datalist = [prepend + str(i + 1) + suffix for i in range(5)]
+    echo_times = [15.4, 29.7, 44.0, 58.3, 72.6]
+    tedana_cli.tedana_workflow(
+        data=datalist,
+        tes=echo_times,
+        ica_method="robustica",
+        n_robust_runs=5,
+        out_dir=out_dir,
+        tedpca=0.95,
+        fittype="curvefit",
+        fixed_seed=49,
+        tedort=True,
+        verbose=True,
+        prefix="sub-01",
+    )
+
+    # Just a check on the component table pending a unit test of load_comptable
+    comptable = os.path.join(out_dir, "sub-01_desc-tedana_metrics.tsv")
+    df = pd.read_table(comptable)
+    assert isinstance(df, pd.DataFrame)
+
+
+def test_integration_four_echo(skip_integration):
+    """Integration test of the full tedana workflow using four-echo test data."""
+
+    if skip_integration:
+        pytest.skip("Skipping four-echo integration test")
+
+    test_data_path, osf_id = data_for_testing_info("four-echo")
+    out_dir = os.path.abspath(os.path.join(test_data_path, "../../outputs/four-echo"))
+    out_dir_manual = f"{out_dir}-manual"
+
+    if os.path.exists(out_dir):
+        shutil.rmtree(out_dir)
```

Comment:
I'm guessing that this is a duplication of existing TEDANA code, in which case the greater fault is said duplication rather than the content of the code itself. But worth mentioning anyway since I'm doing a code review and I see code.

This snippet introduces a race condition whereby if some other process deletes that directory in between the existence test and the deletion, the command will terminate. Better practise in Python is to do:

```
try:
    shutil.rmtree(out_dir)
except FileNotFoundError:
    pass
```

Or:

```
shuteil.rmtree(out_dir, ignore_error=True)
```